Source/Core/VideoCommon/BPMemory.h

```diff
@@ -491,6 +491,81 @@ struct fmt::formatter<TevStageCombiner::ColorCombiner>
   template <typename FormatContext>
   auto format(const TevStageCombiner::ColorCombiner& cc, FormatContext& ctx)
   {
+    auto out = ctx.out();
+    if (cc.bias != TevBias::Compare)
+    {
+      // Generate an equation view, simplifying out addition of zero and multiplication by 1
+      // dest = (d (OP) ((1 - c)*a + c*b) + bias) * scale
+
+      // Note that this is simplified with regards to how lerping works; see PixelShaderGen's
+      // WriteTevRegular for more details.
+
+      static constexpr Common::EnumMap<const char*, TevColorArg::Zero> alt_names = {
+          "prev.rgb", "prev.aaa", "c0.rgb",  "c0.aaa",  "c1.rgb", "c1.aaa", "c2.rgb",    "c2.aaa",
+          "tex.rgb",  "tex.aaa",  "ras.rgb", "ras.aaa", "1",      ".5",     "konst.rgb", "0",
+      };
+
+      const bool has_d = cc.d != TevColorArg::Zero;
+      // If c is one, (1 - c) is zero, so (1-c)*a is zero
+      const bool has_a = cc.a != TevColorArg::Zero && cc.c != TevColorArg::One;
+      // If either b or c is zero, b*c is zero
+      const bool has_b = cc.b != TevColorArg::Zero && cc.c != TevColorArg::Zero;
+      const bool has_bias = cc.bias != TevBias::Zero;  // != Compare is already known
+      const bool has_scale = cc.scale != TevScale::Scale1;
+
+      const char op = (cc.op == TevOp::Sub ? '-' : '+');
+
+      if (cc.dest == TevOutput::Prev)
+        out = format_to(out, "dest.rgb = ");
+      else
+        out = format_to(out, "{:n}.rgb = ", cc.dest);
+
+      if (has_scale)
+        out = format_to(out, "(");
+      if (has_d)
+        out = format_to(out, "{}", alt_names[cc.d]);
+      if (has_a || has_b)
+      {
+        if (has_d)
+          out = format_to(out, " {} ", op);
+        else if (cc.op == TevOp::Sub)
+          out = format_to(out, "{}", op);
+      }
+      if (has_a)
+      {
+        if (cc.c == TevColorArg::Zero)
+          out = format_to(out, "{}", alt_names[cc.a]);
+        else if (cc.c == TevColorArg::Half)
+          out = format_to(out, ".5*{}", alt_names[cc.a]);
+        else
+          out = format_to(out, "(1 - {})*{}", alt_names[cc.c], alt_names[cc.a]);
+      }
+      if (has_a && has_b)
+        out = format_to(out, " {} ", op);
+      if (has_b)
+      {
+        if (cc.c == TevColorArg::One)
+          out = format_to(out, "{}", alt_names[cc.b]);
+        else
+          out = format_to(out, "{}*{}", alt_names[cc.c], alt_names[cc.b]);
+      }
+      if (has_bias)
+      {
+        if (has_a || has_b || has_d)
+          out = format_to(out, cc.bias == TevBias::AddHalf ? " + .5" : " - .5");
+        else
+          out = format_to(out, cc.bias == TevBias::AddHalf ? ".5" : "-.5");
+      }
+      else
+      {
+        // If nothing has been written so far, add a zero
+        if (!(has_a || has_b || has_d))
+          out = format_to(out, "0");
```

Comment:
I don't think that situation is common enough to worry about and would result in a fair bit of additional complexity; similarly, it would be possible to eliminate the parentheses if there's only one term, but that would make the code a lot more messy.